backend/Recodo.BLL/Services/UserService.cs

```diff
@@ -35,6 +43,90 @@ public async Task<UserDTO> CreateUser(NewUserDTO userRegisterDTO)
             return _mapper.Map<UserDTO>(userEntity);
         }
 
+        public async Task UpdateUser(UpdateUserDTO userDto, IFormFile avatar)
+        {
+            var userEntity = await _context.Users.FirstOrDefaultAsync(u => u.Id == userDto.Id);
+            if (userEntity == null) { return; }
+
+            userEntity.WorkspaceName = userDto.WorkspaceName ?? userEntity.WorkspaceName;
+
+
+            if (avatar != null)
+            {
+                using (var client = new HttpClient())
+                {
+                    var url = "https://upload.gyazo.com/api/upload?access_token=" + _configuration["GyazoKey"];
+
+                    using (var memoryStream = new MemoryStream())
+                    {
+                        await avatar.CopyToAsync(memoryStream);
+                        var avatarImage = memoryStream.ToArray();
+
+                        ByteArrayContent byteContent = new ByteArrayContent(avatarImage);
+                        var multipartContent = new MultipartFormDataContent();
+                        multipartContent.Add(byteContent, "imagedata", "imagedata");
+
+                        var response = await client.PostAsync(url, multipartContent);
+                        var stringResponse = await response.Content.ReadAsStringAsync();
+
+                        AvatarDTO json = JsonSerializer.Deserialize<AvatarDTO>(stringResponse);
+                        userEntity.AvatarLink = json.thumb_url;
+                    }
+
+                    //userDto.Avatar
+                }
+            }
+
+            _context.Users.Update(userEntity);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task UpdateUserPassword(UpdateUserDTO userDto)
+        {
+            var userEntity = await _context.Users.FirstOrDefaultAsync(u => u.Id == userDto.Id);
+            if (userEntity == null) { return; }
+
+            if (!SecurityHelper.IsValidPassword(userEntity.Password, userDto.PasswordCurrent, userEntity.Salt))
+            {
+                throw new InvalidUserNameOrPasswordException();
+            }
+
+            userEntity.Email = userDto.Email ?? userEntity.Email;
```

Comment:
Why are you updating password if this method is clearly for updating just the password?